### ISSUE TITLE
[bug]: shell scripts assume bash

### ISSUE THREAD

**Issue by danielb2** [stuartcryan/lastpass-alfred-workflow#16]
Should specify the shell to be bash in the shell scripts with `#!/bin/bash`
For those of us who run [other shells](http://fishshell.com/) lplogin etc is broken due to this.

**Comment by gilesbutler** [comment]
I use ZSH, I presume this is related:
````
$lpass_exec login -trust "$login_email" && clear && echo 'Forcing vault synchronisation... Please wait... (this window will close when complete)' && $lpass_exec ls --sync=now > /dev/null 2>&1 && exit 0

login: illegal option -- t

usage: login [-pq] [-h hostname] [username]
````

**Comment by danielb2** [comment]
@gilesbutler  I don't think so. Looks to me like `$lpass_exec` isn't being set:

``` bash
[~]$ login -t
login: illegal option -- t
```

**Comment by gilesbutler** [comment]
Ah ok thanks @danielb2 any idea on how to set that?

**Comment by danielb2** [comment]
@gilesbutler not sure but it's not being set in the script. I think zsh is close enough to bash where there shouldn't be any syntax issues in setting it, but I don't have the context now

**Comment by stuartcryan** [comment]
I will have a look at making this change to specify the shell properly (normally I always would but for some reason overlooked it in the context of scripts used in Alfred).

Also I will have a squiz at zsh and see if there is anything not working correctly and keep you all posted.

**Comment by gilesbutler** [comment]
Thanks for your help @danielb2 and @stuartcryan

**Comment by stuartcryan** [comment]
Hi Giles and Daniel,
I have been working on a rather large major update which I hope will fix this problem (and in fact should... as I have changed over to using AppleScript for the most part so should behave a bit better.

Just wanted to keep you in the loop, once a beta verison is ready I will post in here so that you can download, have a play and let me know if it has resolved the issues for you.
Cheers,
Stuart

**Comment by danielb2** [comment]
sounds good ;)

**Comment by gilesbutler** [comment]
Awesome stuff, thanks @stuartcryan

**Comment by stuartcryan** [comment]
This has now been done and the new version is released :) this should no longer be an issue.

**Comment by danielb2** [comment]
@stuartcryan still having issues. I deleted the workflow from alfred and reinstalled it using the one in the [latest release zip file](https://github.com/stuartcryan/lastpass-alfred-workflow/archive/1.4.2.zip).

```
doif test -x $f; thenlpass_exec=$ffidoneclear$lpass_exec login --trust "$login_email" && clear && echo 'Forcing vault synchronisation... Please wait... (this window will close when complete)' && $lpass_exec ls --sync=now > /dev/null 2>&1 && exit 0
Unsupported use of '&&'. In fish, please use 'COMMAND; and COMMAND'.
```

it's in the terminal command section of the workflow. what am I doing wrong?

**Comment by danielb2** [comment]
actually, it looks like the .workflow file didn't get updated to 1.4  and it's still 1.2 like I suspected (I just checked the git repo)

**Comment by danielb2** [comment]
![stuartcryan_lastpass-alfred-workflow](https://cloud.githubusercontent.com/assets/26678/12330471/33fb1d5e-ba9a-11e5-8829-df69762ddf50.png)

**Comment by danielb2** [comment]
out of curiosity, is there any way to get this workflow going without using that updated export? If there is, I couldn't find it.

**Comment by stuartcryan** [comment]
Argggh totally my bad all.. I exported but somehow missed it from the update.. SO 1.4.3 has now been released at https://github.com/stuartcryan/lastpass-alfred-workflow/blob/master/lastpass-cli-alfred-workflow.alfredworkflow?raw=true

ANDDDDD as an apology (and because I had a stroke of genius in between). 1.4.3 fixes both the binary and also makes a change to make copying the password to the clipboard LIGHTENING fast :) you should now find that happens almost instantaneously.

Please let me know how you go (and I am re-opening this until I hear from you).

**Comment by danielb2** [comment]
Now it works :) thanks!

**Comment by stuartcryan** [comment]
Schweeeet :) glad to hear it... OK now that the emergency stuff is dealt with I can have a shower, breakfast and get ready to work :: #iFeelDumbThisMorning but hey...you all got an extra (and most awesome) additional feature so easy come easy go I guess.

Have a good one and please pop back and let me know your thoughts overall (tis nice to get feedback on the good, the bad, and the ugly).
Stu

**Comment by gilesbutler** [comment]
Hey Stu,
Awesome work unfortunately for me though it gets stuck when I put my vault password in. It just keeps asking for my password over and over again. I'm pretty certain its the correct password as I logged out of LP and back in again and it works.

Any ideas?

Thanks

**Comment by stuartcryan** [comment]
Hmm that is an odd one just in transit at the moment can you please log a new github issue and I will send you some things you can do to find the issue. 
Cheers
Stuart

This has been sent from my iPhone, apologies for any brevity.

	
Stuart Ryan
BScDipInfTechProfPrac UTS, AACS
IT professional who employs exceptional leadership & interpersonal skills to inspire organisational innovation. 
Connect on LinkedIn | Connect on Twitter - @StuartCRyan | Web: http://stuartryan.com 
I live my life by three mottos:
1. It doesn't matter if you are laughing at me or with me... as long as you're laughing. | 2. I might have to grow old, but I will be damned if I ever have to grow up. | 3. If I can end each and every day having put at least one smile on another person's face, it has been a smashing day!


> On 15 Jan 2016, at 8:41 AM, Giles Butler <notifications@github.com> wrote:
> 
> Hey Stu,
> Awesome work unfortunately for me though it gets stuck when I put my vault password in. It just keeps asking for my password over and over again. I'm pretty certain its the correct password as I logged out of LP and back in again and it works.
> 
> Any ideas?
> 
> Thanks
> 
> —
> Reply to this email directly or view it on GitHub.
>
 
--------------------------------------------------------------------------
This email has been scanned for spam and viruses by a SpamTitan anti-spam appliance. For more information please visit http://stu.link/SPAMTitan

We use Turnkey Internet for all our hosting needs. http://stu.link/TurnkeyInternet

**Comment by gilesbutler** [comment]
Thanks Stu, no rush!

The new issue is [here](https://github.com/stuartcryan/lastpass-alfred-workflow/issues/18).
